src/python/pants/backend/project_info/dependencies.py

```diff
@@ -80,10 +80,11 @@ async def list_dependencies_as_json(
         )
 
         iterated_targets = []
-        for transitive_targets in transitive_targets_group:
-            iterated_targets.append(
-                sorted([str(tgt.address) for tgt in transitive_targets.dependencies])
-            )
+        for idx, transitive_targets in enumerate(transitive_targets_group):
+            targets_collection = {str(tgt.address) for tgt in transitive_targets.dependencies}
+            if dependencies_subsystem.closed:
+                targets_collection.add(addresses[idx].spec)
+            iterated_targets.append(sorted(targets_collection))
```

Comment:
Consider using the `TransitiveTargets.closure` instead of taking the long route of finding the root address used in the `TransitiveTargetsRequest`.

```suggestion
            iterated_targets.append(
              sorted([
                str(tgt.address) for tgt in (
                  transitive_targets.closure if dependencies_subsystem.closed else transitive_targets.dependencies
                )
              ])
            )
```

Reply:
Oh this is great, thanks!